src/main/java/org/folio/inventory/dataimport/handlers/matching/AbstractMatchEventHandler.java

```diff
@@ -63,12 +70,58 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
   }
 
   private CompletableFuture<Boolean> doMatching(DataImportEventPayload dataImportEventPayload, MappingMetadataDto mappingMetadataDto,
-                                                MatchingParametersRelations matchingParametersRelations) {
+                                                MatchingParametersRelations matchingParametersRelations, Context context) {
     dataImportEventPayload.getContext().put(MAPPING_PARAMS, mappingMetadataDto.getMappingParams());
     dataImportEventPayload.getContext().put(MATCHING_RELATIONS,
       Json.encode(matchingParametersRelations.getMatchingRelations()));
 
-    return MatchingManager.match(dataImportEventPayload);
+    return MatchingManager.match(dataImportEventPayload)
+      .thenCompose(matchedLocal -> {
+        if (isConsortiumAvailable()) {
+          return matchCentralTenantIfNeeded(dataImportEventPayload, matchedLocal, context, mappingMetadataDto, matchingParametersRelations);
+        }
+        return CompletableFuture.completedFuture(matchedLocal);
+      });
+  }
+
+  private CompletableFuture<Boolean> matchCentralTenantIfNeeded(DataImportEventPayload dataImportEventPayload, boolean isMatchedLocal, Context context,
+                                                                MappingMetadataDto mappingMetadataDto, MatchingParametersRelations matchingParametersRelations) {
+    return consortiumService.getConsortiumConfiguration(context)
+      .toCompletionStage().toCompletableFuture()
+      .thenCompose(consortiumConfiguration -> {
+        if (consortiumConfiguration.isPresent() && !consortiumConfiguration.get().getCentralTenantId().equals(context.getTenantId())) {
+          LOGGER.debug("matchCentralTenantIfNeeded:: Start matching on central tenant with id: {}", consortiumConfiguration.get().getCentralTenantId());
+          String localMatchedInstance = dataImportEventPayload.getContext().get(getEntityType().value());
+          preparePayloadBeforeConsortiumProcessing(dataImportEventPayload, consortiumConfiguration.get(), mappingMetadataDto, matchingParametersRelations);
+          return MatchingManager.match(dataImportEventPayload)
+            .thenCompose(isMatchedConsortium -> {
+              dataImportEventPayload.setTenant(context.getTenantId());
+              if (isMatchedConsortium && isMatchedLocal) {
+                LOGGER.warn("matchCentralTenantIfNeeded:: Found multiple results during matching on local tenant: {} and central tenant: {} ",
+                  consortiumConfiguration.get().getCentralTenantId(), context.getTenantId());
```

Comment:
I think there is a case where a shared instance is matched in the central tenant and the corresponding shadow instance in matched in the local tenant. So it is possible to have multiple results and still be fine. What do you think?

Reply:
Hi, Olamide, thanks, this is a good point, I added a check if the Instance matched locally is a shadow instance, if yes we do not start the consortium matching process

Reply:
As discussed in our meeting today, more changes are going to be made?

Reply:
Added this changes, please review